03-Modify-Raster.Rmd

```diff
@@ -74,197 +72,240 @@ Read more about the `raster` package in R.</a>
+1) by subtracting the two rasters in `R` 
+or if our rasters are large
+2) using the `overlay()` function to more efficiently calculate the difference.
+
+First, we'll learn the intuitive first step of subtracting layers to create and
+plot a `CHM`.  In the geospatial world we often call
+this *raster math*.
+
+If you completed Lessons 00 and 02 you've already loaded and viewed 
+`HARV_dtmCrop.tif` and `HARV_dsmCrop.tif` and can skip this code chunk.  If you 
+have not completed it, let's load and check out our raster files. 
 ```{r load-libraries }
-
 #load raster package
 library(raster)
 
-#view info about the dtm raster data
+#view info about the dtm & dsm raster data
 GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#load the DTM
+GDALinfo("NEON_RemoteSensing/HARV/DTM/HARV_dsmCrop.tif")
+#load the DTM & DSM rasters
 DTM <- raster("NEON_RemoteSensing/HARV/DTM/HARV_dtmCrop.tif")
-
-#create a quick plot
+DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
+#create a quick plot of each to see what we're dealing with
 plot(DTM,
      main="Digital Terrain Model (Elevation)\n Harvard Forest")
 
-#If it isn't already loaded, load the DSM as well 
-DSM <- raster("NEON_RemoteSensing/HARV/DSM/HARV_dsmCrop.tif")
-
+plot(DSM,
+     main="Digital Surface Model (Elevation)\n Harvard Forest")
 ```
 
 Once the data are loaded, we can use mathametical operations on them.
-For instance, we can subtract the DTM from the DSM to create a `Canopy Height
-Model`.
+For instance, we can subtract the DTM from the DSM to create a Canopy Height
+Model.
 
 ```{r raster-math }
 # Raster math example
-
-CHM <- DSM - DTM #This section could be automatable later on
+CHM <- DSM - DTM 
 
 #view the output CHM
 plot(CHM,
      main="NEON Canopy Height Model - Subtracted\n Harvard Forest") 
 ```
-
-#Exploring Raster Data Values
-
-It's often a good idea to explore the range of values in a raster dataset just like
-we might explore a dataset that we collected in the field. We can do this using
-the `hist` function in `R`.
-
-What range of values do you expect to see for the Harvard Forest `Canopy Height 
-Model (CHM)` that we just created?
-
-
-```{r view-histogram }
-
-#Create histogram of CHM values
+Notice the legend on the right.  We are now looking at the height of the canopy 
+instead of the elevation hence the smaller numbers.  
+
+#Challenge: Exploring Raster Values in CHM
+As we learned in previous lessons, it's often a good idea to explore the range 
+of values in a raster dataset just like we might explore a dataset that we 
+collected in the field.  
+
+1. What range of values do you expect to see for the Harvard Forest Canopy 
+Height Model (`CHM`) that we just created? 
+2. What are two ways you can check this range of data in CHM? 
+3. Plot your data in such a way that you can see the range of data values in 
+CHM.
+4. Bonus challenge: Change the color of and add a title to the plot you created 
+in question 3. 
+
+```{r challenge-code-1 }
```

Comment:
Done to all in Raster Lessons 00-04.  Will apply to others throughout workshop as I encounter them.